Plugins/Creature/Creature.cpp

```diff
@@ -1047,12 +1047,26 @@ NWNX_EXPORT ArgumentStack SetClassByPosition(ArgumentStack&& args)
     {
         const auto position = args.extract<int32_t>();
         const auto classID = args.extract<int32_t>();
+        const auto bUpdateLevels = args.extract<int32_t>();
+        const auto classIDold = pCreature->m_pStats->GetClass(position);
```

Comment:
This line should move below the ASSERT_OR_THROWs, so that if `position` is not valid, we get a throw and not a potential crash.